### PR TITLE
Trx logger class name fix - NUnit data driven tests

### PR DESCRIPTION
**Pull request by abhishkk**
## Description
Example:
```
        [TestCase(2, 2, 4.0)]
        public void TestMethod1(int a, int b, double c)
        {
            Assert.IsTrue(a + b == c);
        }
```
For above example, trx file has `className` as `UnitTestProject1.UnitTest1.TestMethod1(2, 3, 4.`.
Ideally className should be  `UnitTestProject1.UnitTest1`

## Related issue
https://github.com/Microsoft/vsts-tasks/issues/7620

### CODE REVIEW

**Review by nigurr — APPROVED** [review]

**Review by nigurr — APPROVED** [review]
Assuming you have written all possible tests for this

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs [hunk 1]
```diff
@@ -572,45 +572,46 @@ private static string GetOwner(ObjectModel.TestCase rockSteadyTestCase)
...
-                    }
-                }
-                catch (ArgumentException)
-                {
-                    // If source is not valid file path, then className will continue to point default value.
+                    return testCaseSource;
                 }
             }
+            catch (ArgumentException)
+            {
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/Converter.cs]: Add logging here.

File: test/Microsoft.TestPlatform.Extensions.TrxLogger.UnitTests/Utility/ConverterTests.cs [hunk 2]
```diff
@@ -60,6 +64,80 @@ public void ToCollectionEntriesShouldRenameAttachmentUriIfTheAttachmentNameIsSam
...
+
+            var unitTestElement = Converter.ToTestElement(testCase.Id, Guid.Empty, Guid.Empty, testCase.DisplayName, TrxLoggerConstants.UnitTestType, testCase);
+
+            object[] expected = Enumerable.Empty<Object>().ToArray();
+
+            CollectionAssert.AreEqual(expected, unitTestElement.TestCategories.ToArray());
+        }
+
+        [TestMethod]
+        public void ToTestElementShouldContainExpectedTestMethodPropertiesIfFqnEndsWithTestName()
```
Comment (smadala) [inline test/Microsoft.TestPlatform.Extensions.TrxLogger.UnitTests/Utility/ConverterTests.cs]: DRY; use datarow here.

**Review by smadala — APPROVED** [review]

**Comment by abhishkk** [comment]
@dotnet-bot test this please

**Comment by abhishkk** [comment]
@dotnet-bot test this please.

**Comment by abhishkk** [comment]
@dotnet-bot test Windows_NT / Debug Build

**Comment by abhishkk** [comment]
@dotnet-bot test Windows_NT / Release Build

**Comment by abhishkk** [comment]
@dotnet-bot test Windows_NT / Debug Build

**Pull request merged**